app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt

```diff
@@ -448,9 +451,7 @@ class LocalLibraryFragment : BaseFragment() {
       } else if (!isPermissionDenied(grantResults)) {
         fragmentDestinationLibraryBinding?.apply {
           permissionDeniedLayoutShowing = false
-          fileManagementNoFiles.visibility = GONE
-          goToDownloadsButtonNoFiles.visibility = GONE
-          zimfilelist.visibility = VISIBLE
+          requestFileSystemCheck()
```

Comment:
Why are we removing it here and adding that in the check permission? Plus we are checking `requestFileSystemCheck` twice which is before the permission and after the permission. Is it necessary?  It's computationally expensive.

Reply:
> Plus we are checking requestFileSystemCheck twice which is before the permission and after the permission. Is it necessary? It's computationally expensive.

good catch! i had forget to remove this.

> Why are we removing it here and adding that in the check permission?

I have added the visibility for views after ```requestFileSystemCheck``` method so it's easy to maintain views according to filesystem.